File: app/src/main/java/com/elegion/android/template/ui/base/activity/BaseActivity.java
package com.elegion.android.template.ui.base.activity;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.Nullable;

import com.arellomobile.mvp.MvpAppCompatActivity;

import butterknife.ButterKnife;

/**
 * @author mikhail.funikov
 */
public abstract class BaseActivity extends MvpAppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayout());
        ButterKnife.bind(this);
    }

    @LayoutRes
    protected abstract int getLayout();
}

File: app/src/main/kotlin/com/bankex/pay/ui/base/activity/BaseActivity.kt
package com.bankex.pay.ui.base.activity

import android.os.Bundle
import android.support.annotation.LayoutRes
import com.arellomobile.mvp.MvpAppCompatActivity

abstract class BaseActivity : MvpAppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(getLayout())
    }

    @LayoutRes
    protected abstract fun getLayout(): Int
}
